src/main/scala/com/fulcrumgenomics/personal/nhomer/HasSequence.scala

```diff
@@ -0,0 +1,157 @@
+/**
+  * Copyright (c) 2016, Fulcrum Genomics LLC
+  * All rights reserved.
+  *
+  * Redistribution and use in source and binary forms, with or without
+  * modification, are permitted provided that the following conditions are met:
+  *
+  * 1. Redistributions of source code must retain the above copyright notice,
+  * this list of conditions and the following disclaimer.
+  *
+  * 2. Redistributions in binary form must reproduce the above copyright notice,
+  * this list of conditions and the following disclaimer in the documentation
+  * and/or other materials provided with the distribution.
+  *
+  * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+  * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+  * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+  * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+  * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+  * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+  * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+  * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+  * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+  * POSSIBILITY OF SUCH DAMAGE.
+  */
+
+package com.fulcrumgenomics.personal.nhomer
+
+import java.nio.file.Path
+
+import com.fulcrumgenomics.cmdline.{JeanLucTool, Personal}
+import dagr.commons.CommonsDef.PathToBam
+import dagr.commons.io.Io
+import dagr.sopt._
+import htsjdk.samtools._
+import htsjdk.samtools.metrics.{MetricBase, MetricsFile}
+import htsjdk.samtools.util._
+import picard.PicardException
+
+import scala.io.Source
+
+@clp(
+  description = "Searches for DNA sequences in the read pairs.",
+  group = classOf[Personal]
+)
+class HasSequence
+( @arg(doc = "Input SAM or BAM.") val input: PathToBam,
+  @arg(doc = "Output SAM or BAM.") val output: PathToBam,
+  @arg(doc = "File containing the DNA sequences to search for.") val sequences: Path,
+  @arg(doc = "Sequence match histogram written to this file.") val metrics: Path,
+  @arg(doc = "Maximum mismatches for matching a sequence.") val maxMismatches: Int = 1,
+  @arg(doc = "Minimum base quality. Any bases falling below this quality will be considered a mismatch even in the bases match.", flag = "q") val minimumBaseQuality: Int = 0,
+  @arg(doc = "Treat no calls in the input sequences as DNA bases.") val noCallsAreBases: Boolean = false,
+  @arg(doc = "Search for mismatches at the start only") val startOnly: Boolean = true,
+  @arg(doc = "The tag to store the result.") val hasSequenceTag: String = "XW"
+) extends JeanLucTool {
+  Io.assertReadable(input)
```

Comment:
Would be nice to tag options as `@in` and `@out` and have JeanLuc take care of performing these checks for us.  Or we could have annotations for `@r @w @rw`?